features/cells/src/main/java/com/wire/android/feature/cells/ui/versionhistory/VersionHistoryViewModel.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.feature.cells.ui.versionhistory
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class VersionHistoryViewModel @Inject constructor(
+    val savedStateHandle: SavedStateHandle,
+) : ViewModel() {
+
+    fun getVersionsGroupedByDate(): Map<String, List<CellVersion>> {
+        // todo: implement real logic to get versions in next PR
+        return mapOf("1 Dec 2025" to listOf())
+    }
```

Comment:
Suggestion: Just to not forget later, we can from now start using immutable maps here to help compose, even is not relevant for this dummy data.

```suggestion
    fun getVersionsGroupedByDate(): ImmutableMap<String, List<CellVersion>> {
        // todo: implement real logic to get versions in next PR
        return  persistentMapOf("1 Dec 2025" to listOf())
    }
    
    
```